core/src/main/java/org/infinispan/persistence/spi/CacheWriter.java

```diff
@@ -35,4 +38,28 @@
     * @throws PersistenceException in case of an error, e.g. communicating with the external storage
     */
    boolean delete(Object key);
+
+   /**
+    * Persist all provided entries to the store in a single batch update. If this is not supported by the
+    * underlying store, then entries are written to the store sequentially via {@link #write(MarshalledEntry)}
```

Comment:
Please document if the list may contain duplicate entries for the same key (in that case we really need a list, otherwise it could be just a Collection).